actionpack/lib/action_dispatch/middleware/host_authorization.rb

```diff
@@ -137,16 +142,24 @@ def call(env)
         mark_as_authorized(request)
         @app.call(env)
       else
+        @response_app.blocked_hosts = blocked_hosts(request) if @response_app.respond_to?(:"blocked_hosts=")
         @response_app.call(env)
       end
     end
 
     private
-      def authorized?(request)
+      def blocked_hosts(request)
         origin_host = request.get_header("HTTP_HOST")
         forwarded_host = request.x_forwarded_host&.split(/,\s?/)&.last
 
-        @permissions.allows?(origin_host) && (forwarded_host.blank? || @permissions.allows?(forwarded_host))
+        hosts = []
+        hosts << origin_host unless @permissions.allows?(origin_host)
+        hosts << forwarded_host unless forwarded_host.blank? || @permissions.allows?(forwarded_host)
+        hosts
```

Comment:
I think using `reject` here makes the code simpler:
```suggestion
        [origin_host, forwarded_host].compact_blank.reject {|host|  @permissions.allows?(host) }
```

Reply:
awesome, thanks!